presto-main/src/main/java/io/prestosql/operator/PipelineContext.java

```diff
@@ -197,19 +197,7 @@ public void driverFinished(DriverContext driverContext)
         // merge the operator stats into the operator summary
         List<OperatorStats> operators = driverStats.getOperatorStats();
         for (OperatorStats operator : operators) {
-            // TODO: replace with ConcurrentMap.compute() when we migrate to java 8
-            OperatorStats updated;
-            OperatorStats current;
-            do {
-                current = operatorSummaries.get(operator.getOperatorId());
-                if (current != null) {
-                    updated = current.add(operator);
-                }
-                else {
-                    updated = operator;
-                }
-            }
-            while (!compareAndSet(operatorSummaries, operator.getOperatorId(), current, updated));
+            operatorSummaries.compute(operator.getOperatorId(), (k, v) -> v == null ? operator : v.add(operator));
```

Comment:
You should use https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#merge-K-V-java.util.function.BiFunction- instead

Reply:
Done